### DISCUSSION TITLE
An option to backup the file on track.save();

### DISCUSSION THREAD

**Discussion by DividedSEv2** [Zeugma440/atldotnet#113] [Ideas]
I dont think this is a issue rather a discussion so i didnt create a issue.
Toying with Tags is too much pain while backing up track multiple time for testing purpose... 
Would be great if track.save(bool doBackup(default false)) on each save with timestamp (track12102021.mp3) only for faster development & testing.
Do you think it is viable option for atl or atl clients do need it itself?

**Comment by Zeugma440** [comment]
In my humble opinion, from an API design point of view, having that option would look really weird.
I haven't seen any equivalent library that has anything similar.

Maybe you could get inspiration from what I've set up for my own testing app : 

Sample code (from `ATL.test.IO.HighLevel.TagIO_RW_DeleteTag`)
```cs
// Create a temp copy of the given reference file
string testFileLocation = TestUtils.CopyAsTempTestFile("MP3/01 - Title Screen.mp3");
// Instanciate the Track from the temp copy
Track theTrack = new Track(testFileLocation);

// Here be some tests

// Get rid of the temp copy
if (Settings.DeleteAfterSuccess) File.Delete(testFileLocation);
```

A simplified version of TestUtils is there :
https://github.com/Zeugma440/atldotnet/blob/master/ATL.benchmark/TestUtils.cs

The code creates the temp copy into the project's `Resources\tmp` folder but you can adapt it to do whatever you need.

If you test the features of an actual app by using its GUI, you may want to play with `#if DEBUG` (e.g. if `DEBUG` is set, work on a temp copy; if not, work on the actual file)

**Comment by Zeugma440** [comment]
>  As u were curious about my project earlier, if u want to see i can attach some screenshot of what i done with atl here.

Pretty please, go ahead 😄

**Reply by DividedSEv2** [reply]
Ok 👌 I will post some soon.

**Comment by Mapaler** [comment]
I'm following the tutorial.
The parameters of `theTrack.Save();` do not contain FilePath or Stream.

I want to save the modified mp3 as a new file, and I don't need to constantly restore the original file that hasn't been modified。

 I'd like to be able to refer to design of `System.Drawing.Image.Save()`
```C#
Track.SaveAs(Stream stream);
Track.SaveAs(string Path);
```

**Reply by Zeugma440** [reply]
I like that suggestion - the `System.Drawing.Image` API looks nice indeed. I'll consider it for the next updates and keep you informed

**Reply by Zeugma440** [reply]
`Track.SaveAs` (and its `async` variants) are available on today's v5.26

**Reply by Mapaler** [reply]
> `Track.SaveAs` (and its `async` variants) are available on today's v5.26

I tried using v5.26 but VS got an error.

![devenv_Audi_MMI_MP3_Cover_Repair_-_Microsoft_Visual_Studi_0342](https://github.com/user-attachments/assets/1bca5355-b02d-499a-b91b-f58bf345a140)

![devenv_Audi_MMI_MP3_Cover_Repair_-_Microsoft_Visual_Studi_0344](https://github.com/user-attachments/assets/c0ff7d54-f2db-491f-9a33-137ab8023d5c)

**Reply by Zeugma440** [reply]
Whoops... the name of the new method is `SaveTo` and not `SaveAs` 😅 
Sorry for the confusion
